include/mbedtls/net_sockets.h

```diff
@@ -106,6 +106,16 @@ mbedtls_net_context;
  */
 void mbedtls_net_init( mbedtls_net_context *ctx );
 
+/**
+ * \brief          This function sets the file descriptor.
+ *
+ * \param ctx      The context to configure.
+ * \param fd       The file descriptor.
+ *
+ * \return         O if succesful, or MBEDTLS_ERR_NET_INVALID_CONTEXT
+ */
```

Comment:
Add a comment that fd set by this function will be overwritten by `mbedtls_net_connect`, `mbedtls_net_bind`, `mbedtls_net_accept`, `mbedtls_net_close` and `mbedtls_net_free`

Suggested:
```
 * \note           File descriptor set by this function will be overwritten by
 *                 calling any of the the following functions:
 *                 \c mbedtls_net_connect(), \c mbedtls_net_bind(), \c mbedtls_net_accept()
 *                 \c mbedtls_net_close() and \c mbedtls_net_free()
```

Reply:
It's even more than that: calling `net_close()` or `net_free()` will also change the state of the file descriptor. Suggested documentation:

```
 * \note           The file descriptor set by this function will be overwritten by
 *                 calling any of the the following functions:
 *                 mbedtls_net_connect(), mbedtls_net_bind(), mbedtls_net_accept()
 *                 mbedtls_net_close() and mbedtls_net_free().
 *                 Additionally, mbedtls_net_close() and mbedtls_net_free() will alter the state
 *                 of the file descriptor before overwriting it.
```
(Note: we don't need `\c` in front of function names - doxygen will recognize them anyway.)

Reply:
+1 for \c, didn't know that :)